src/native/libs/System.Native/pal_uid.c

```diff
@@ -237,3 +237,55 @@ int32_t SystemNative_GetGroups(int32_t ngroups, uint32_t* groups)
 
     return getgroups(ngroups, groups);
 }
+
+int32_t SystemNative_GetUName(uint32_t uid, char* buffer, size_t bufferSize)
+{
+    assert(buffer != NULL);
+    assert(bufferSize > 0);
+
+    if (buffer == NULL || bufferSize <= 0)
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    struct passwd u;
+    struct passwd* pu; 
+
+    errno = 0;
+    int e;
+    while ((e = getpwuid_r(uid, &u, buffer, bufferSize, &pu)) == EINTR);
+
+    if (e < 0)
+    {
+        errno = e;
+        return -1;
+    }
+    return (int32_t)strlen(buffer);
+}
+
+int32_t SystemNative_GetGName(uint32_t gid, char* buffer, size_t bufferSize)
+{
+    assert(buffer != NULL);
+    assert(bufferSize > 0);
+
+    if (buffer == NULL || bufferSize <= 0)
+    {
+        errno = EINVAL;
+        return -1;
+    }
+
+    struct group g;
+    struct group* pg; 
+
+    errno = 0;
+    int e;
+    while ((e = getgrgid_r(gid, &g, buffer, bufferSize, &pg)) == EINTR);
```

Comment:
We need to access `gr_name` to get the name.

Reply:
I tested this in a console C++ app. The group name was saved in the buffer.

Reply:
I'll do what you suggest for both cases.